### PR TITLE
Layout update, auto tracking bug fixes

### PR DESCRIPTION
**Pull request by haate**
Updating layout for locations to include Cardia Incentive, fix the munged logic on the Cardia Incentive, modified tail logic to show completed if 'Fight Bahamut' flag is checked

Layout change is as follows:
image

Moved the shop item to the last line, added in the cardia incentive where it was.

The logic change on the tail is now using the tail's status as well as that of Bahamut (I also corrected the spelling in that function so it's `bahamut` as opposed to `bahumat` now). Thankfully enough, Big B's status changes depending on if he's killed or if the trade is made. If the trade is made, his status goes to 3.0, and if he's murdered dead, it goes to 2. Rather than using negations on the status, I put them to use the specific status numbers, and using both statuses.

### CODE REVIEW

File: scripts/autotracking.lua [hunk 1]
```diff
@@ -501,7 +503,7 @@ function updateLocationsFromMemorySegmentCorridor(segment)
         updateSectionMultipleChestCountFromByteAndFlag(segment, "@Cardia Grassy/Chests", { [0x628C] = 0x04, [0x628D] = 0x04, [0x628E] = 0x04 })
         updateSectionMultipleChestCountFromByteAndFlag(segment, "@Cardia Swampy/Chests", { [0x6289] = 0x04, [0x628A] = 0x04, [0x628B] = 0x04 })
         updateSectionMultipleChestCountFromByteAndFlag(segment, "@Cardia Forest/Chests", { [0x6284] = 0x04, [0x6285] = 0x04, [0x6286] = 0x04, [0x6288] = 0x04, [0x628F] = 0x04, [0x6290] = 0x04 })
-		updateSectionMultipleChestCountFromByteAndFlag(segment, "@Cardia Forest/Incentive", 0x6287, 0x04)
+		updateSectionSingleChestCountFromByteAndFlag(segment, "@Cardia Forest/Cardia Incentive", 0x6287, 0x04)
```
Comment (openhouse-jtoyoda) [inline scripts/autotracking.lua:506]: Looks like the spacing is funny here

File: scripts/autotracking.lua [hunk 2]
```diff
@@ -160,15 +160,17 @@ function updateTail(segment)
...
+            print(item.Name, tail, bahamut, item.CurrentStage)
         end
-
-        if tail > 0 then
+		
+        if tail > 0 and bahamut == 1 then
             item.CurrentStage = 1
-        elseif bahumat & 0x02 ~= 0 then
-            item.CurrentStage = 2
+		elseif bahamut == 2 and tail == 1 then
```
Comment (openhouse-jtoyoda) [inline scripts/autotracking.lua:170]: This should use an 'OR' rather than two branches if it means the same thing. Also a comment would be helpful

**Review by openhouse-jtoyoda — CHANGES_REQUESTED** [review]

File: scripts/autotracking.lua (hunk 1 above)
Comment (haate) [reply scripts/autotracking.lua:506]: Damnable thing used actual tabs as opposed to matching to space format like it bloody well should have. Fixing that up now. Will be addressing it (and the below) in the next merge.

File: scripts/autotracking.lua (hunk 2 above)
Comment (haate) [reply scripts/autotracking.lua:170]: See above comment. Working on fix now. Next merge will address.

File: scripts/autotracking.lua [hunk 3]
```diff
@@ -160,14 +160,14 @@ function updateTail(segment)
...
         if AUTOTRACKER_ENABLE_DEBUG_LOGGING then
-            print(item.Name, ruby, bahumat, item.CurrentStage)
+            print(item.Name, tail, bahamut, item.CurrentStage)
         end
 
-        if tail > 0 then
+        if tail > 0 and bahamut == 1 then
             item.CurrentStage = 1
-        elseif bahumat & 0x02 ~= 0 then
+        elseif bahamut == 2 and tail == 1 or bahamut == 3 and tail == 0 then
```
Comment (openhouse-jtoyoda) [inline scripts/autotracking.lua:170]: This logic is incorrect. It needs brackets

**Review by openhouse-jtoyoda — CHANGES_REQUESTED** [review]

**Comment by haate** [comment]
Alright, wasn't aware of that syntax with lua scripting. I've been just doing code monkey stuff.

Hopefully I interpreted your meaning correctly.

**Comment by haate** [comment]
Discovered that killing a patrolling warmech *also* updates Bahamut's status ID, and could trigger the tail to mark as completed even if the tail hadn't been found. Reworked the originalBahamut status logic to only trigger once the tail was picked up. May still lead to false positives if wandering warmech is killed before tail is turned in, but unable to test that edge case currently.

Also discovered that the enable location tracking preference seems to not actually function, as it updates the location whether or not you've got that flag set to true or false, but I have *no clue* where to start troubleshooting that issue.

**Comment by jtoyoda** [comment]
> Discovered that killing a patrolling warmech _also_ updates Bahamut's status ID, and could trigger the tail to mark as completed even if the tail hadn't been found. Reworked the originalBahamut status logic to only trigger once the tail was picked up. May still lead to false positives if wandering warmech is killed before tail is turned in, but unable to test that edge case currently.
> 
> Also discovered that the enable location tracking preference seems to not actually function, as it updates the location whether or not you've got that flag set to true or false, but I have _no clue_ where to start troubleshooting that issue.

Oh that is something that is core to the pack

File: scripts/autotracking.lua [hunk 4]
```diff
@@ -156,21 +156,30 @@ function updateSlab(segment)
...
+originalBahamut = ""
 function updateTail(segment)
     local item = Tracker:FindObjectForCode("tail")
     if item then
         local tail = ReadU8(segment, 0x602D)
-        local bahumat = ReadU8(segment, 0x620E)
-        if AUTOTRACKER_ENABLE_DEBUG_LOGGING then
-            print(item.Name, ruby, bahumat, item.CurrentStage)
+        local bahamut = ReadU8(segment, 0x620E)
+		if AUTOTRACKER_ENABLE_DEBUG_LOGGING then
```
Comment (jtoyoda) [inline scripts/autotracking.lua:165]: Looks like the wrong spacing again

File: scripts/autotracking.lua [hunk 5]
```diff
@@ -156,21 +156,30 @@ function updateSlab(segment)
...
+            print(item.Name, tail, bahamut, item.CurrentStage, hadTail, originalBahamut)
         end
-
-        if tail > 0 then
+
```
Comment (jtoyoda) [inline scripts/autotracking.lua:168]: Extra white space

File: scripts/autotracking.lua [hunk 6]
```diff
@@ -156,21 +156,30 @@ function updateSlab(segment)
...
+			hadTail = true
+		end
+		if ( originalBahamut == "" and hadTail == true ) then
+				originalBahamut = bahamut
+		end
+		
+        if ( bahamut ~= originalBahamut and tail == 1 ) or ( bahamut ~= originalBahamut and hadTail == true ) or ( hadTail == true and tail == 0 ) then
+			item.CurrentStage = 2
+		elseif hadTail == true then
+			item.CurrentStage = 1
```
Comment (jtoyoda) [inline scripts/autotracking.lua:180]: Spacing looks wrong here

File: scripts/autotracking.lua [hunk 7]
```diff
@@ -156,21 +156,30 @@ function updateSlab(segment)
     end
 end
 
+originalBahamut = ""
```
Comment (jtoyoda) [inline scripts/autotracking.lua:159]: This should go up with the other globals. Also it should default to a hex number?

File: scripts/autotracking.lua [hunk 8]
```diff
@@ -156,21 +156,30 @@ function updateSlab(segment)
...
+				originalBahamut = bahamut
+		end
+		
+        if ( bahamut ~= originalBahamut and tail == 1 ) or ( bahamut ~= originalBahamut and hadTail == true ) or ( hadTail == true and tail == 0 ) then
+			item.CurrentStage = 2
+		elseif hadTail == true then
+			item.CurrentStage = 1
         elseif AUTOTRACKER_ENABLE_SETTING_LOCATIONS_TO_FALSE then
-          item.CurrentStage = 0
+			item.CurrentStage = 0
```
Comment (jtoyoda) [inline scripts/autotracking.lua:182]: Spacing is wrong here

**Review by jtoyoda — CHANGES_REQUESTED** [review]

File: scripts/autotracking.lua (hunk 7 above)
Comment (haate) [reply scripts/autotracking.lua:159]: This one, I left as a null value until the first time the tail is picked up to ensure that it would a) have a semi-fresh value to compare against and b) wouldn't return any value until after it's actually propagated. This was done to prevent any potential misfires. If you'd prefer, I can set it to `false` and get the same effect.

**Comment by haate** [comment]
The one conversation I commented on is the only thing left. the rest were all tabs being filled in rather than the editor using double spaces. Fixed that up with a simple sed. 
I gotta remember to check for that ere I commit. Maybe set up a git hook for my local....

**Comment by jtoyoda** [comment]
> The one conversation I commented on is the only thing left. the rest were all tabs being filled in rather than the editor using double spaces. Fixed that up with a simple sed.
> I gotta remember to check for that ere I commit. Maybe set up a git hook for my local....

I don't see a commit but once these are ready to go we can merge

**Comment by haate** [comment]
yeah, i hadn't pushed as I was waiting for final clarification on what you'd prefer the initial value for the originalBahamut variable to be. Currenlty, it's still a nil value, but has been moved up to be with the rest of the global variable declarations.

**Review by jtoyoda — APPROVED** [review]

**Comment by jtoyoda** [comment]
> yeah, i hadn't pushed as I was waiting for final clarification on what you'd prefer the initial value for the originalBahamut variable to be. Currenlty, it's still a nil value, but has been moved up to be with the rest of the global variable declarations.

I think -1 is correct as a default value. People were just asking for this so I want to get this merged in asap

**Comment by haate** [comment]
Changed and pushed!
